fe/fe-core/src/main/java/org/apache/doris/datasource/ExternalCatalog.java

```diff
@@ -240,6 +240,20 @@ public List<String> getDbNames() {
         return listDatabaseNames(null);
     }
 
+    @Override
+    public List<String> getDbNamesOrEmpty() {
+        if (initialized) {
```

Comment:
it is strange because in `getDbNames()`, it will try to initialize it and then get databases.
if we add `if (initialized)` here, it will not do initialization anymore.

Reply:
> 

Because we don't want to initialize when querying the information_schema table, an uninitialized catalog can easily cause the query to time out